### PR TITLE
Rework config system

### PR DESCRIPTION
**Pull request by Heath123**
Cleans up the config system slightly, and also fixes launch.json on Linux because I couldn't test otherwise (it should still work on Windows too)

### CODE REVIEW

File: bot_modules/config_manager/index.js [hunk 1]
```diff
@@ -45,7 +45,7 @@ exports.init = function init (client) {
       if (args[1] === 'edit') {
         if (args[2] === 'all') {
           try {
-            configEditor.set(JSON.parse(args.slice(3).join(' ')))
+            exports.configEditor.set(JSON.parse(args.slice(3).join(' ')))
```
Comment (rtm516) [inline bot_modules/config_manager/index.js:48]: Could `exports.configEditor` not be replaced with `this`

File: bot_modules/dump_analyse/index.js [hunk 2]
```diff
@@ -3,7 +3,7 @@ const GitHub = require('github-api')
 const ipRangeCheck = require('ip-range-check')
 const mcping = require('mcping-js')
 
-let config = {}
+const config = require('../config_manager/index.js')
```
Comment (rtm516) [inline bot_modules/dump_analyse/index.js:6]: This should use `const { configEditor} = ...` so you don't have to do `config.configEditor`

File: bot_modules/config_manager/index.js (hunk 1 above)
Comment (Heath123) [reply bot_modules/config_manager/index.js:48]: Not exactly, because this code isn't part of `configEditor`, so `this` isn't `configEditor`

Comment (rtm516) [reply bot_modules/config_manager/index.js:48]: The current just doesn't seem clean, but likely the best way. Thanks for checking.

File: bot_modules/dump_analyse/index.js (hunk 2 above)
Comment (Heath123) [reply bot_modules/dump_analyse/index.js]: Done

**Pull request merged**